File: TimeSheetManager/TimeSheetManager.BLL/Service/EmployeeService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TimeSheetManager.BLL.DTO;
using TimeSheetManager.BLL.IService;
using TimeSheetManager.DAL.Repository;
using TimeSheetManager.Models;

namespace TimeSheetManager.BLL.Service
{
    public class EmployeeService : IEmployeeService
    {
        private readonly IEmployeeRepository _repo;

        public EmployeeService(IEmployeeRepository repo)
        {
            _repo = repo;
        }

        public EmployeeProfileDTO GetProfile(string username)
        {
            var emp = _repo.GetByUsername(username);
            if (emp == null) throw new Exception("Employee not found");

            return new EmployeeProfileDTO
            {
                EmployeeId = emp.EmployeeId,
                FullName = emp.FullName,
                Email = emp.Email,
                Department = emp.Department,
                Position = emp.Position,
                Status = emp.Status,
                Username = emp.User?.Username
            };
        }

        public void UpdateProfile(string username, UpdateProfileRequest request)
        {
            var emp = _repo.GetByUsername(username);
            if (emp == null) throw new Exception("Employee not found");

            // Update allowed fields
            emp.FullName = request.FullName;
            emp.Department = request.Department;
            emp.Position = request.Position;

            _repo.Update(emp);
        }
    }
}

File: fe-timesheetmanager/src/app/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface EmployeeProfile {
  employeeId: number;
  fullName: string;
  email: string;
  department: string;
  position: string;
  status: string;
  username: string;
}

export interface UpdateProfileRequest {
  fullName: string;
  department: string;
  position: string;
}

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  private apiUrl = 'https://localhost:7156/api/employee';

  constructor(private http: HttpClient) {}

  getProfile(): Observable<EmployeeProfile> {
    return this.http.get<EmployeeProfile>(`${this.apiUrl}/profile`);
  }

  updateProfile(data: UpdateProfileRequest): Observable<any> {
    return this.http.put(`${this.apiUrl}/profile`, data);
  }
}
